aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -240,11 +263,12 @@ Tensor stft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop
   auto window_ = window;
   if (win_length < n_fft) {
     // pad center
-    window_ = at::zeros({n_fft}, self.options());
     auto left = (n_fft - win_length) / 2;
     if (window.defined()) {
+      window_ = at::zeros({n_fft}, window.options());
       window_.narrow(0, left, win_length).copy_(window);
     } else {
+      window_ = at::zeros({n_fft}, self.options());
```

Comment:
Narrowing and filling in this case seems odd. Would it be possible to init the window using `full` and avoid narrowing by computing its size?